Bluetooth: Fix double set random address for every advertising enable

LingaoM opened the pull request:
For Bluetooth Mesh, all Mesh messages will use private address by default, and then will call `bt_le_ext_adv_update_param`
for Mesh message, this function will call `bt_id_set_adv_own_addr` to set controller random address

But when call `bt_le_ext_adv_start` to start advertising, will also call `bt_id_set_adv_private_addr` to set new random address, that is duplicated, BTW, also increase latency to send mesh message, because every hci command/event complete/status need time.

This PR also fix two bbsim test problem:
1. Mesh relay, look like timing changed after this PR, not real problem.
2. PAST, look like address in `ADV_EXT_IND`, `AUX_ADV_IND` shall not same with `ADV_SYNC_IND`;

LingaoM committed:
Bluetooth: Host: Fix unnecessary random address update

The `bt_le_ext_adv_update_param` will set new random address when option
not select `BT_LE_ADV_OPT_USE_IDENTITY`, But `bt_le_ext_adv_start` will
also set random address again.

This will be affect bluetooth mesh, after this change, will ensure address
only change once for every advertising.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

LingaoM committed:
Bluetooth: Host: Fix Periodic Advertising random address update

update random address when periodic adv enabled.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

jhedberg commented:
@LingaoM looks generally fine, but could you please add more info to the third commit message, i.e. what exactly was the timing problem and why is your fix the correct one.

jhedberg commented:
(I updated the PR title to reflect that this is a bug fix, thereby making it possible to consider it post-rc1)

jhedberg commented:
@LingaoM and regarding the two host commit messages, could you use consistent prefixes please, e.g. "Bluetooth: Host: Fix unnecessary random address update" and "Bluetooth: Host: Fix Periodic Advertising random address update"

LingaoM commented:
``` C
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/zephyr  ‹random_address_set_twice*› 
╰─➤  RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml \
   SEARCH_PATH=${ZEPHYR_BASE}/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh  \
      ${ZEPHYR_BASE}/tests/bsim/run_parallel.sh
Attempting to run 1 cases (logging to  /media/mi/5e58d875-5e34-6b43-8936-d956a2f60c04/works/github/zephyrproject/zephyr/myresults.xml)
/home/mi/works/github/zephyrproject/zephyr/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh PASSED
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/zephyr  ‹random_address_set_twice*› 
╰─➤  
```

@jhedberg bsim for bap_bass_client_sync.sh failed, but i'm local passed.

LingaoM commented:
> > ```c
> > (.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/zephyr  ‹random_address_set_twice*› 
> > ╰─➤  RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml \
> >    SEARCH_PATH=${ZEPHYR_BASE}/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh  \
> >       ${ZEPHYR_BASE}/tests/bsim/run_parallel.sh
> > Attempting to run 1 cases (logging to  /media/mi/5e58d875-5e34-6b43-8936-d956a2f60c04/works/github/zephyrproject/zephyr/myresults.xml)
> > /home/mi/works/github/zephyrproject/zephyr/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh PASSED
> > (.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/zephyr  ‹random_address_set_twice*› 
> > ╰─➤  
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @jhedberg bsim for bap_bass_client_sync.sh failed, but i'm local passed.
> 
> @LingaoM, it failed on my machine the same way as in CI.

<img width="2078" height="574" alt="image" src="https://github.com/user-attachments/assets/616dad74-c36f-4aab-9d4b-64d74d3d0fb3" />
Fantastic, is there anything else I overlooked?

PavelVPV commented:
> > > Fantastic, is there anything else I overlooked?
> > 
> > 
> > Maybe you need to update and rebuild [babblesim components](https://github.com/zephyrproject-rtos/zephyr/blob/8aba2e3f3aa87deb9c8606ef54fddca1d936d031/west.yml#L36-L118)?
> 
> 1. west config manifest.group-filter -- +babblesim
>    west update
> 2. west update
>    cd /home/mi/works/github/zephyrproject/tools/bsim; make everything -j 8
> 3. export BSIM_OUT_PATH=${ZEPHYR_BASE}/../tools/bsim
>    export BSIM_COMPONENTS_PATH=${BSIM_OUT_PATH}/components/
> 
> <img alt="image" width="2000" height="574" src="https://private-user-images.githubusercontent.com/26378606/504676219-bfdb92c8-c0c7-486a-85ef-9029437ef001.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyMTU1OTcsIm5iZiI6MTc2MTIxNTI5NywicGF0aCI6Ii8yNjM3ODYwNi81MDQ2NzYyMTktYmZkYjkyYzgtYzBjNy00ODZhLTg1ZWYtOTAyOTQzN2VmMDAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDIzVDEwMjgxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1NTEyNjVkMzRkMzNhODU1MTU5ZmU0NDE2ZjkxMzVmYzg2NWYxMjdhNzIzMmI4NGUwN2Q0NGNiMDA4NDdhYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mK-r-f9H5555Q3KN9FEJy_VMDaNU6Vi5yZeHMQwbkCY">

Are you running it for `nrf5340bsim/nrf5340/cpuapp` target?

LingaoM commented:
> Are you running it for `nfr5340bsim/nrf5340/cpuapp` target?

``` C
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/tools/bsim  ‹2ba22a0› 
╰─➤  ${ZEPHYR_BASE}/tests/bsim/compile.sh       
Building /home/mi/works/github/zephyrproject/zephyr/../tools/bsim/bin/bs_nrf52_bsim_native_tests_bsim_bluetooth_audio_prj_conf
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/tools/bsim  ‹2ba22a0› 
╰─➤  RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml \
   SEARCH_PATH=${ZEPHYR_BASE}/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh   \
      ${ZEPHYR_BASE}/tests/bsim/run_parallel.sh
Attempting to run 1 cases (logging to  /media/mi/5e58d875-5e34-6b43-8936-d956a2f60c04/works/github/zephyrproject/zephyr/myresults.xml)
/home/mi/works/github/zephyrproject/zephyr/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh PASSED
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/tools/bsim  ‹2ba22a0› 
```

```
diff --git a/tests/bsim/bluetooth/compile.sh b/tests/bsim/bluetooth/compile.sh
index f8e62a7fb63..0ac7123ed23 100755
--- a/tests/bsim/bluetooth/compile.sh
+++ b/tests/bsim/bluetooth/compile.sh
@@ -16,14 +16,6 @@ source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
 # On the other hand the audio compile script, only builds one image. So we parallelize it with
 # the rest to save a couple of seconds.
 run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/audio/compile.sh
-${ZEPHYR_BASE}/tests/bsim/bluetooth/audio_samples/compile.sh
-${ZEPHYR_BASE}/tests/bsim/bluetooth/host/compile.sh
-${ZEPHYR_BASE}/tests/bsim/bluetooth/ll/compile.sh
-${ZEPHYR_BASE}/tests/bsim/bluetooth/mesh/compile.sh
-${ZEPHYR_BASE}/tests/bsim/bluetooth/samples/compile.sh
-if [ ${BOARD} == "nrf52_bsim/native" ]; then
-	${ZEPHYR_BASE}/tests/bsim/bluetooth/hci_uart/compile.sh
-fi
-${ZEPHYR_BASE}/tests/bsim/bluetooth/tester/compile.sh
+
 
 wait_for_background_jobs
diff --git a/tests/bsim/bluetooth/ll/compile.sh b/tests/bsim/bluetooth/ll/compile.sh
index d5058386824..15333c91f47 100755
--- a/tests/bsim/bluetooth/ll/compile.sh
+++ b/tests/bsim/bluetooth/ll/compile.sh
@@ -10,20 +10,6 @@ set -ue
 
 source ${ZEPHYR_BASE}/tests/bsim/compile.source
 
-${ZEPHYR_BASE}/tests/bsim/bluetooth/ll/cis/compile.sh
-
-app=tests/bsim/bluetooth/ll/advx compile
-app=tests/bsim/bluetooth/ll/advx \
-  conf_overlay=overlay-ticker_expire_info.conf compile
-app=tests/bsim/bluetooth/ll/advx conf_overlay=overlay-scan_aux_use_chains.conf compile
-
-app=tests/bsim/bluetooth/ll/conn conf_file=prj_split.conf compile
-app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_1ms.conf compile
-app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_tx_defer.conf compile
-app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_privacy.conf compile
-app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_low_lat.conf compile
-app=tests/bsim/bluetooth/ll/conn conf_file=prj_split_single_timer.conf compile
-
 app=tests/bsim/bluetooth/ll/bis conf_overlay=overlay-sequential.conf compile
 app=tests/bsim/bluetooth/ll/bis conf_overlay=overlay-interleaved.conf  compile
 app=tests/bsim/bluetooth/ll/bis conf_overlay=overlay-ll_interface.conf compile
@@ -32,15 +18,4 @@ app=tests/bsim/bluetooth/ll/bis conf_overlay=overlay-scan_aux_use_chains.conf co
 app=tests/bsim/bluetooth/ll/bis conf_file=prj_vs_dp.conf compile
 app=tests/bsim/bluetooth/ll/bis conf_file=prj_past.conf compile
 
-app=tests/bsim/bluetooth/ll/edtt/hci_test_app \
-  conf_file=prj_dut_llcp.conf compile
-app=tests/bsim/bluetooth/ll/edtt/hci_test_app \
-  conf_file=prj_tst_llcp.conf compile
-app=tests/bsim/bluetooth/ll/edtt/gatt_test_app \
-  conf_file=prj_llcp.conf compile
-
-app=tests/bsim/bluetooth/ll/multiple_id compile
-app=tests/bsim/bluetooth/ll/throughput compile
-app=tests/bsim/bluetooth/ll/throughput conf_overlay=overlay-no_phy_update.conf compile
-
 wait_for_background_jobs

diff --git a/tests/bsim/compile.sh b/tests/bsim/compile.sh
index 3b1a7dce635..8945f6e981d 100755
--- a/tests/bsim/compile.sh
+++ b/tests/bsim/compile.sh
@@ -12,6 +12,5 @@ set -ue
 source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
 
 run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/compile.sh
-run_in_background ${ZEPHYR_BASE}/tests/bsim/net/compile.sh
 
 wait_for_background_jobs

```

LingaoM commented:
> Didn't get this. You need to set `BOARD` variable before test runs:

``` C
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/zephyr  ‹random_address_set_twice*› 
╰─➤                                                                                                                                                                        130 ↵
(.venBOARD=nrf5340bsim/nrf5340/cpuapp ${ZEPHYR_BASE}/tests/bsim/compile.sh                                                                                                 130 ↵
Building /home/mi/works/github/zephyrproject/zephyr/../tools/bsim/bin/bs_nrf5340bsim_nrf5340_cpuapp_tests_bsim_bluetooth_audio_prj_conf
(.venv-3.10) ╭─mi@lingao ~/works/github/zephyrproject/zephyr  ‹random_address_set_twice*› 
╰─➤  BOARD=nrf5340bsim/nrf5340/cpuapp RESULTS_FILE=${ZEPHYR_BASE}/myresults.xml \
   SEARCH_PATH=${ZEPHYR_BASE}/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh   \
      ${ZEPHYR_BASE}/tests/bsim/run_parallel.sh
Attempting to run 1 cases (logging to  /media/mi/5e58d875-5e34-6b43-8936-d956a2f60c04/works/github/zephyrproject/zephyr/myresults.xml)
/home/mi/works/github/zephyrproject/zephyr/tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh FAILED


======== Running BASS Client Sync =========

d_02: @00:00:00.000000  (CPU:0): *** Booting Zephyr OS build v4.2.0-6751-g31898f5fa543 ***
d_01: @00:00:00.000000  (CPU:0): *** Booting Zephyr OS build v4.2.0-6751-g31898f5fa543 ***
d_00: @00:00:00.000000  (CPU:0): *** Booting Zephyr OS build v4.2.0-6751-g31898f5fa543 ***
d_02: @00:00:00.000000  (CPU:0): [00:00:00.000,000] <dbg> bt_ots: bt_gatt_ots_l2cap_init: Initialized OTS L2CAP
d_01: @00:00:00.000000  (CPU:0): [00:00:00.000,000] <dbg> bt_ots: bt_gatt_ots_l2cap_init: Initialized OTS L2CAP
d_00: @00:00:00.000000  (CPU:0): [00:00:00.000,000] <dbg> bt_ots: bt_gatt_ots_l2cap_init: Initialized OTS L2CAP
d_02: @00:00:00.000000  (CPU:1): *** Booting Zephyr OS build v4.2.0-6751-g31898f5fa543 ***
d_00: @00:00:00.000000  (CPU:1): *** Booting Zephyr OS build v4.2.0-6751-g31898f5fa543 ***
d_01: @00:00:00.000000  (CPU:1): *** Booting Zephyr OS build v4.2.0-6751-g31898f5fa543 ***
```

LingaoM committed:
tests: bluetooth: audio: fix bsim timing

move extending advertising start after periodic adv.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

PavelVPV committed:
bluetooth: mesh: adv_ext: Fix scheduling with multiple relay adv sets

When multiple relay adv sets are used, the bt_mesh_adv_send function
calls bt_mesh_adv_relay_ready which should distribute relayed
advertisements across all relay adv sets.

Until the submitted relay adv set work is started, the ADV_FLAG_ACTIVE
is not set. Therefore, next call to bt_mesh_adv_send will try to
re-submit the same relay adv set work, instead of picking up another
relay set which is actually free and ready to send an advertisement.

This commit adds a check that checks if the adv set work is already
pending to be executed. And if so, schedule_send returns false to make
bt_mesh_adv_relay_ready pick next relay adv set.

This shouldn't brake advertising because once adv set is done
transmitting advertisment, it will pick up a next one.

The ADV_FLAG_PROXY check is added to do re-submit for adv set which was
used for proxy advertisement since we need to prioritize mesh messages
over proxy advertisements when those are running on the same adv set.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

codecoup-tester commented:
Scheduled PR https://github.com/zephyrproject-rtos/zephyr/pull/98117#issuecomment-3437446522, board: nrf54, estimated start time: 16:52:22, test case count: 10, estimated duration: 0:13:01<details><summary>Test cases to be run</summary>GAP/PADV/PASM/BV-01-C<br>GAP/PADV/PAM/BV-01-C<br>GAP/PADV/PASE/BV-01-C<br>GAP/PADV/PASE/BV-02-C<br>GAP/PADV/PASE/BV-03-C<br>GAP/PADV/PASE/BV-04-C<br>GAP/PADV/PASE/BV-05-C<br>GAP/PADV/PASE/BV-06-C<br>GAP/PADV/PAST/BV-01-C<br>GAP/PADV/PAST/BV-02-C</details>

PavelVPV reviewed (approved):
Approving mesh change and host change (as much as I can see it right now, though can miss something).
